metals/src/main/scala/scala/meta/internal/metals/ClientExperimentalCapabilities.scala

```diff
@@ -1,58 +1,87 @@
 package scala.meta.internal.metals
-import com.google.gson.JsonElement
-import com.google.gson.JsonNull
+import com.google.gson.JsonObject
 import org.eclipse.{lsp4j => l}
 
+/**
+ * While all these values can be set here, they are here only for
+ * compatablity with clients that were setting them this way. From
+ * a client perspective it's preferable and recommended to use
+ * InitializationOptions instead. From a development perspective
+ * don't add something here unless it's truly a more "experimental"
+ * type feature.
+ */
 final case class ClientExperimentalCapabilities(
-    debuggingProvider: Boolean,
-    decorationProvider: Boolean,
-    didFocusProvider: Boolean,
-    doctorProvider: String,
-    executeClientCommandProvider: Boolean,
-    inputBoxProvider: Boolean,
-    openFilesOnRenameProvider: Boolean,
-    quickPickProvider: Boolean,
-    slowTaskProvider: Boolean,
-    statusBarProvider: String,
-    treeViewProvider: Boolean
+    debuggingProvider: Option[Boolean],
+    decorationProvider: Option[Boolean],
+    didFocusProvider: Option[Boolean],
+    doctorProvider: Option[String],
+    executeClientCommandProvider: Option[Boolean],
+    inputBoxProvider: Option[Boolean],
+    openFilesOnRenameProvider: Option[Boolean],
+    quickPickProvider: Option[Boolean],
+    slowTaskProvider: Option[Boolean],
+    statusBarProvider: Option[String],
```

Comment:
https://github.com/scalameta/metals/blob/48b20201f3357738ce472c40dcae47b0cccc56fb/docs/editors/new-editor.md

Here we say `statusBarProvider` is Boolean :thinking:

Reply:
🤔 that's a mistake in the docs then. Good catch. I'll change this there.

Reply:
Updated this.

Reply:
Changed.